terraform/cp4ba/main.tf

```diff
@@ -0,0 +1,179 @@
+}
+
+resource "null_resource" "mkdir_kubeconfig_dir" {
+  triggers = { always_run = timestamp() }
+  provisioner "local-exec" {
+    command = "mkdir -p ${var.config_dir}"
+  }
+}
+
+data "ibm_container_cluster_config" "cluster_config" {
+  depends_on = [null_resource.mkdir_kubeconfig_dir]
+  cluster_name_id   = var.cluster_name_id
+  resource_group_id = data.ibm_resource_group.group.id
+  download          = true
+  config_dir        = "./kube/config"
+  admin             = false
+  network           = false
+}
+
+
+module "cluster" {
+  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak/tree/main/modules/roks"
+//  source = "../../roks"
+  // source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//modules/roks"
+  enable = local.enable_cluster
+//  on_vpc = var.on_vpc
+
+  project_name             = var.project_name
+  owner                    = var.entitled_registry_user_email
+  environment              = var.environment
+
+  resource_group       = var.resource_group
+  roks_version         = var.platform_version
+  flavors              = var.flavors
+  workers_count        = var.workers_count
+  datacenter           = var.data_center
+  force_delete_storage = true
+  vpc_zone_names       = var.vpc_zone_names
+
+  private_vlan_number  = var.private_vlan_number
+  public_vlan_number   = var.public_vlan_number
+}
+
+# getting and creation a directory for the cluster config file
+resource "null_resource" "mkdir_kubeconfig_dir" {
+  triggers = { always_run = timestamp() }
+    provisioner "local-exec" {
+    command = "mkdir -p ${var.cluster_config_path}"
+  }
+}
+
+//////######################### LDAP ##################################
+module "ldap" {
+  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak/tree/main/modules/ldap"
+//  source = "../../ldap"
+//  source = "git::https://github.com/ibm-hcbt/terraform-ibm-cloud-pak.git//modules/ldap"
+
+  ibmcloud_api_key      = var.ibmcloud_api_key
+  region                = var.region
+  iaas_classic_api_key  = var.iaas_classic_api_key
+  iaas_classic_username = var.iaas_classic_username
+  ssh_public_key_file   = var.ssh_public_key_file
+  ssh_private_key_file  = var.ssh_private_key_file
+  classic_datacenter    = var.classic_datacenter
+}
+
+module "cp4ba"{
+    source = "git::https://github.com/jgod1360/terraform-ibm-cloud-pak/tree/cp4ba/modules/cp4ba"
+//    source = "../.."
+    // TODO: With Terraform 0.13 replace the parameter 'enable' or the conditional expression using 'with_iaf' with 'count'
+    enable = true
+
+    openshift_version   = var.openshift_version
+    cluster_config_path = data.ibm_container_cluster_config.cluster_config.config_dir
+    cluster_name_id     = var.cluster_name_id
+//    on_vpc              = var.on_vpc
+
+    // IBM Cloud API Key
+    ibmcloud_api_key = var.ibmcloud_api_key
+
+    # Cluster
+//    on_vpc                        = var.on_vpc
+    portworx_is_ready             = var.portworx_is_ready
```

Comment:
This should be retrieved from the `portworx` module.  See https://github.com/ibm-hcbt/cloud-pak-sandboxes/blob/master/terraform/cp4data/main.tf, line 104.  Portworx should probably be called before this module is called.